.github/workflows/ci.yml

```diff
@@ -84,19 +96,23 @@ jobs:
         update: true
         install: mingw-w64-x86_64-toolchain mingw-w64-x86_64-boost msys2-runtime-devel git mingw-w64-x86_64-cmake mingw-w64-x86_64-ninja
 
+    - name: Update Rust (UNIX)
+      if: matrix.os != 'windows-latest'
+      run: rustup update
+
     - name: Switch target (Windows)
       if: matrix.os == 'windows-latest'
       run: rustup toolchain install stable-x86_64-pc-windows-gnu -c clippy && rustup set default-host x86_64-pc-windows-gnu && rustup default stable-x86_64-pc-windows-gnu
 
-    - name: Update Rust
-      run: rustup update
+    - name: Documentation
+      run: cargo doc --workspace --all-features
 
     - name: Clippy (fail on warnings)
-      run: cargo clippy --workspace --all-targets --all-features -- -D warnings
+      run: cargo clippy --workspace --all-features -- -D warnings
 
     - name: Test
-      run: cargo test --all-features --workspace --all-targets
+      run: cargo test --all-features --workspace
 
     # TODO: upload binaries with `actions/upload-artifact@v3`
     - name: Build
-      run: cargo build --all-features --all-targets --workspace
```

Comment:
`--all-targets` does not mean all target archs it means tests, benches, libs etc so we should keep it for `clippy` and `build`. For `test` it seems specifying  `--all-targets` acctually reduces the amount of our tests ran: it skips docs tests so we should keep `--all-targets` removed for that